src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -297,6 +299,8 @@ public int StopDebuggingPE()
                 // Avoid ending the debug session if it has already been ended.
                 if (_encService.DebuggingSession != null)
                 {
+                    _encService.OnBeforeDebuggingStateChanged(DebuggingState.Run, DebuggingState.Design);
```

Comment:
Yeah, I'm relying on knowledge that the only valid state transitions are Design ↔ Run ↔ Break. No direct transitions between Design/Break.